src/main/java/com/mcmoddev/alt/AdditionalLootTables.java

```diff
@@ -8,89 +9,100 @@
+import net.minecraftforge.fml.event.server.FMLServerStartingEvent;
+import net.minecraftforge.fml.event.server.FMLServerStoppedEvent;
+import net.minecraftforge.fml.loading.FMLPaths;
+
 
-@Mod(modid = AdditionalLootTables.MODID,
-	version = AdditionalLootTables.VERSION,
-	name = AdditionalLootTables.NAME,
-	acceptedMinecraftVersions = "[1.12,)",
-	certificateFingerprint = "@FINGERPRINT@")
+@Mod(AdditionalLootTables.MODID)
 public class AdditionalLootTables {
-	public static final String NAME = "Additional Loot Tables";
 	public static final String MODID = "alt";
 	public static final String VERSION = "2.0.3";
 
-	@Instance
 	public static AdditionalLootTables INSTANCE = null;
 	public static Config config = null;
 	public static Logger logger = LogManager.getFormatterLogger(MODID);
 	public static final String ALT_FOLDER_NAME = "additional-loot-tables";
 	private static Path lootFolder;
-    private static final String PROXY_BASE = "com.mcmoddev." + MODID + ".proxy.";
     private static final PluginLoader pluginLoader = new PluginLoader();
-    
-    @SidedProxy(clientSide = PROXY_BASE + "ClientProxy", serverSide = PROXY_BASE + "ServerProxy")
-    public static CommonProxy proxy;
 
-	@EventHandler
-	public void onFingerprintViolation(FMLFingerprintViolationEvent event) {
-		logger.warn("Invalid fingerprint detected!");
-	}
+    public static CommonProxy proxy = DistExecutor.runForDist(() -> () -> new ClientProxy(), () -> () -> new CommonProxy());
 
-	@EventHandler
-	public static void preInit(FMLPreInitializationEvent event) {
-		final String ALT_BASE_CONFIG_PATH = event.getSuggestedConfigurationFile().getParent();
-		lootFolder = Paths.get(ALT_BASE_CONFIG_PATH, ALT_FOLDER_NAME);
-		config = new Config(event);
+	public AdditionalLootTables() {
+		if (INSTANCE == null)
+			INSTANCE = this;
+		else
+			throw new RuntimeException("Duplicated Class Instantiation: CustomItems");
+		
+		config = Config.construct();
+		
+		lootFolder = Paths.get(FMLPaths.CONFIGDIR.get().toString(), ALT_FOLDER_NAME);
 		ALTFileUtils.createDirectoryIfNotPresent(lootFolder);
-		pluginLoader.load(event);
-		MinecraftForge.EVENT_BUS.register(ALTEventHandler.class);
 	}
-
+    
+    // TODO: Fix fingerprint detection, FMLFingerprintViolationEvent is not fired anywhere in FML/Forge!
+//	@EventHandler
+//	public void onFingerprintViolation(FMLFingerprintViolationEvent event) {
+//		logger.warn("Invalid fingerprint detected!");
+//	}
+    
+	@Mod.EventBusSubscriber(modid = AdditionalLootTables.MODID, bus = Mod.EventBusSubscriber.Bus.MOD)
+	public static class ModEventHandler {
+    	@SubscribeEvent
+    	public static void newRegistry(RegistryEvent.NewRegistry event) throws IOException {
+    		pluginLoader.load();
+    	}
+		
+		@SubscribeEvent
+		public static void onCommonSetup(FMLCommonSetupEvent event) {
+			pluginLoader.loadResources();
+		}
+		
+		// TODO: fix inter-mod communication
+		@SubscribeEvent

```

Comment:
Thanks for including this - it might be that this is the way forward for the integration...

Reply:
We may have to look into this together, I have never used inter-mod communication in 1.15.2

Reply:
I create an implemention of IMC processing, see the latest commit